packages/discord.js/src/structures/UncachedGuildMember.js

```diff
@@ -0,0 +1,332 @@
+     */
+    this.flags = data.flags;
+
+    /**
+     * Whether this member has yet to pass the guild's membership gate
+     * @type {?boolean}
+     */
+    this.pending = data.pending;
+
+    /**
+     * The total permissions of the member in this channel, including overwrites
+     * @type {Readonly<PermissionsBitField>}
+     */
+    this.parsedPermissions = new PermissionsBitField(data.permissions).freeze();
+
+    /**
+     * Raw total permissions of the member in this channel, including overwrites
+     * @type {?string}
+     * @deprecated Use {@link UncachedGuildMember#parsedPermissions} instead.
+     * This field will be replaced with parsedPermissions in the future.
+     */
+    this.permissions = data.permissions;
+
+    /**
+     * The timestamp this member's timeout will be removed
+     * @type {?number}
+     */
+    this.communicationDisabledUntilTimestamp =
+      data.communication_disabled_until && Date.parse(data.communication_disabled_until);
+
+    /**
+     * The timestamp this member's timeout will be removed, as an ISO8601 timestamp
+     * @type {?string}
+     * @deprecated Use {@link UncachedGuildMember#communicationDisabledUntilTimestamp}
+     * or {@link UncachedGuildMember#communicationDisabledUntil} instead
+     */
+    this.communication_disabled_until = data.communication_disabled_until;
+
+    /**
+     * Whether this UncachedGuildMember is a partial (always true, as it is a partial GuildMember)
+     * @type {boolean}
+     * @readonly
+     */
+    this.partial = true;
+  }
+
+  /**
+   * A link to the member's guild avatar.
+   * @param {ImageURLOptions} [options={}] Options for the image URL
+   * @returns {?string}
+   */
+  avatarURL(options = {}) {
+    return this.avatar && this.client.rest.cdn.guildMemberAvatar(this.guildId, this.id, this.avatar, options);
+  }
+
+  /**
+   * A link to the member's guild avatar if they have one.
+   * Otherwise, a link to their {@link User#displayAvatarURL} will be returned.
+   * @param {ImageURLOptions} [options={}] Options for the Image URL
+   * @returns {string}
+   */
+  displayAvatarURL(options) {
+    return this.avatarURL(options) ?? this.user.displayAvatarURL(options);
+  }
+
+  /**
+   * The time this member joined the guild
+   * @type {?Date}
+   * @readonly
+   */
+  get joinedAt() {
+    return this.joinedTimestamp && new Date(this.joinedTimestamp);
+  }
+
+  /**
+   * The time this member's timeout will be removed
+   * @type {?Date}
+   * @readonly
+   */
+  get communicationDisabledUntil() {
+    return this.communicationDisabledUntilTimestamp && new Date(this.communicationDisabledUntilTimestamp);
+  }
+
+  /**
+   * The last time this member started boosting the guild
+   * @type {?Date}
+   * @readonly
+   */
+  get premiumSince() {
+    return this.premiumSinceTimestamp && new Date(this.premiumSinceTimestamp);
+  }
+
+  /**
+   * The member's id
+   * @type {Snowflake}
+   * @readonly
+   */
+  get id() {
+    return this.user.id;
+  }
```

Comment:
does this need to be a getter. Why can't we just assign it statically as user ids don't change.

Reply:
Because if it isn't a getter you're storing the id twice in each member object